pkg/cli/cmds/server.go

```diff
@@ -212,6 +213,11 @@ var ServerFlags = []cli.Flag{
 		Destination: &ServerConfig.FlannelBackend,
 		Value:       "vxlan",
 	},
+	&cli.BoolFlag{
+		Name:        "multi-cluster-cidr",
```

Comment:
Should this perhaps be prefixed with `flannel-` like the rest of the flannel options?
```suggestion
		Name:        "flannel-multi-cluster-cidr",
```

Reply:
it's not flannel directly related. If you have a CNI that supports this feature you can enable it and deploy K3s with `--flannel-backend=none`.

Reply:
Hmm. I guess that's fair. The non-flannel bits are just rbac and a couple extra flags to enable the feature-gate, correct? If that's the case then at some point those bits will move out of alpha and become enabled by default, at which point all we'll have behind this flag is Flannel setup, which is why it feels like a flannel option.

Reply:
Yes it's correct, but the final implementation will require also the possibility to add different CIDRs from different IP families. Right now we are limited to the IP family used at cluster startup.

Reply:
Would it be possible to have Flannel auto-detect support for this by searching for `ClusterCIDR` resources? It feels kind of odd that this requires manual enabling when one of the major enhancements of MultiClusterCIDR is that the CIDRs are now directly discoverable as cluster resources.

Having the Kubernetes stuff behind a flag while it is in Alpha makes sense as, but we need to plan now for this flag to be deprecated when upstream moves the FeatureGate into Beta and it is on by default. Having flannel auto-detect the feature, and having the flag just control the Kubernetes feature config, would make that easier.

Reply:
Yes I am agree with you. Right now you have to enable both the network alpha api and the `MultiCIDRRangeAllocator` and even if the `ClusterCIDR` resource is enabled if the right allocator wasn't configured it's not rightly populated.